gui/elements/_layer_list.py

```diff
@@ -297,30 +310,74 @@ def _add(self, event):
         """Callback when an item is added to set its order and viewer.
         """
         layer = event.item
-        self._qt.layersList.insert(event.index, len(self), layer)
         layer._order = -len(self)
         layer.viewer = self.viewer
+        layer.events.select.connect(self.viewer._update_active_layers)
+        layer.events.deselect.connect(self.viewer._update_active_layers)
 
     def _remove(self, event):
         """Callback when an item is removed to remove its viewer
         and reset its order.
         """
         layer = event.item
-        self._qt.layersList.remove(layer)
         layer.viewer = None
         layer._order = 0
+        layer.events.select.disconnect(self.viewer._update_active_layers)
+        layer.events.deselect.disconnect(self.viewer._update_active_layers)
 
     def _reorder(self, event):
         """Callback when the list is reordered to propagate those changes
         to the node draw order.
         """
         for i in range(len(self)):
             self[i]._order = -i
-        self._qt.layersList.reorder()
         canvas = self.viewer._canvas
         canvas._draw_order.clear()
         canvas.update()
 
+    def _insert_reorder(self, index, insert):
+        """Reorder list by moving the item at index and insterting it
+        at the insert index. If additional items are selected these will
+        get inserted at the insert index too. This allows for rearranging
+        the list based on dragging and dropping a selection of items, where
+        index is the index of the primary item being dragged, and insert is
+        the index of the drop location, and the selection indicates if
+        multiple items are being dragged.
+
+        Parameters
+        ----------
+        index : int
+            Index of primary item to be moved
+        insert : int
+            Index that item(s) will be inserted at
+        """
+        total = len(self)
+        indices = list(range(total))
+        if self[index].selected:
+            selected = []
+            for i in range(total):
+                if self[i].selected:
+                    selected.append(i)
+        else:
+            selected = [index]
+        for i in selected:
+            indices.remove(i)
+        offset = sum([i < insert for i in selected])
+        j = insert - offset
+        for i in selected:
+            indices.insert(j, i)
+            j = j+1
+        if not self[index].selected:
+            self.unselect_all()
+            self[index].selected = True
+        if not indices == [i for i in range(total)]:
+            self.reorder(indices)
```

Comment:
I also don't understand why this is needed...

Reply:
i've removed it - it was to prevent calling the reorder when nothing would actually get reordered (i.e. to save time). Instead now I've prevented this whole function from being called when nothing would get reordered (i.e. if `index == insert` or `index+1 == insert`) - which makes more sense to me. I can add it back in if desired but it i think it is fine now